src/playlist.js

```diff
@@ -235,9 +235,7 @@ const getPlaylistSyncPoints = function(playlist) {
   if (!playlist || !playlist.segments) {
     return [null, null];
   }
-
-  let expiredSync = playlist.syncInfo || null;
-
+  let expiredSync = playlist.endList ? { time: 0, mediaSequence: 0} : playlist.syncInfo || null;
```

Comment:
When a live stream is finished, usually an `EXT-X-ENDLIST` tag is applied to the end of the playlist even though it wasn't originally a VOD. This would set `expiredSync` to `{ time: 0, mediaSequence: 0 }` even if that isn't an actual sync point. We should prioritize `playlist.syncInfo` if it exists, otherwise `{ time: 0, mediaSequence: 0 }` if `playlist.endList` is true, otherwise `null`.